src/osmf/lib/HDSPlayer.as

```diff
@@ -750,7 +750,7 @@ package
         totalTime = 0;
       }
       eventObject.currentTime = _mediaPlayerSprite.mediaPlayer.currentTime;
-      eventObject.duration = duration
+      eventObject.duration = totalTime;
```

Comment:
I tested with live streams, of course, it works fine.  To see it working, you would have to use branch  pbw-5275 - fix for this bug enables this fix